common/djangoapps/microsite_configuration/views.py

```diff
@@ -2,38 +2,101 @@
 """
 Views for microsites api end points.
 """
-import sys, yaml
+import cStringIO  
 from collections import OrderedDict
+import copy
+import logging
+from PIL import Image
+import sys, yaml
+import urllib2
+import urlparse 
+
 from django.apps import apps
 from django.conf import settings
 from django.contrib.sites.models import Site
+from django.core.files.base import ContentFile, File
+from django.db import IntegrityError
+from django.shortcuts import get_object_or_404
+
 from edx_rest_framework_extensions.authentication import JwtAuthentication
+
 from rest_framework.response import Response
 from rest_framework import status
-from rest_framework.viewsets import ViewSet
-from django.shortcuts import get_object_or_404
 from rest_framework.authentication import SessionAuthentication
 from rest_framework.permissions import IsAuthenticated
 from rest_framework_oauth.authentication import OAuth2Authentication
-from microsite_configuration.models import Microsite, MicrositeOrganizationMapping
-from microsite_configuration.serializers import MicrositeSerializer, MicrositeModelSerializer
+from rest_framework.viewsets import ViewSet
+
+from microsite_configuration.models import \
+    Microsite, MicrositeOrganizationMapping
+
+from microsite_configuration.serializers import \
+    MicrositeSerializer, MicrositeModelSerializer
+
 from organizations import api as organizations_api
 from organizations import exceptions as org_exceptions
-from openedx.core.djangoapps.user_api.errors import UserNotFound, UserNotAuthorized
+from organizations.models import Organization
 
-"""
-    The following 3 methods are helper functions to:
-        1: generate a proper json object for the microsite values to be sent up and down the server
-        2: generate a yaml file that gets saved on the server that will be used to generate,
-           configuration files for the microsites to run through lms and for the nginx site configurations.
-           
-    The file location is controlled via EDXAPP_MICROSITE_CONFIG_FILE kept in server_vars.yml
+from openedx.core.djangoapps.user_api.errors import \
+    UserNotFound, UserNotAuthorized
+
+log = logging.getLogger(__name__)
 
+def download_image(url):
+    """
+    Downloads an image
+    Returns a PIL Image, otherwise raises an exception.
+    """
+    r = urllib2.Request(url)
+    try:
+        request = urllib2.urlopen(r, timeout=10)
+        image_data = cStringIO.StringIO(request.read()) # StringIO is a file
```

Comment:
Please just make double sure you don't have to explicitly close the StringIO object after using it, i know that in other languages such as java and c# you have to close it or it will create a memory leak.

Reply:
I don't think so, no examples of the code call the .close() method. I create a temp file and the write it and save which I think deals with the buffer and mem leaks.